pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -46,19 +51,28 @@ func NewPlaceholderPodController(manager ctrl.Manager, conf *config.Config, ingr
 }
 
 func (p *PlaceholderPodController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var err error
+	result := ctrl.Result{}
+
+	// do metrics
+	defer func() {
+		//placing this call inside a closure allows for result and err to be bound after Reconcile executes
+		//this makes sure they have the proper value
+		//just calling defer metrics.HandleControllerReconcileMetrics(controllerName, result, err) would bind
+		//the values of result and err to their zero values, since they were just instantiated
+		metrics.HandleControllerReconcileMetrics(placeholderPodControllerName, result, err)
+	}()
+
 	logger, err := logr.FromContext(ctx)
 	if err != nil {
-		return ctrl.Result{}, err
+		return result, err
 	}
 	logger = logger.WithName("placeholderPodController")
```

Comment:
The logger name can be derived from the controller name so if we ever need to correlate through some kind of query we can. This is beneficial because we add extra details to ICMs created by these metrics that will query for controller logs (filtered through logger name).

Maybe the logger name for each controller is the controller name just shifted from snake case to camel case? We  currently have no contract with anything in terms of logger names so the logger names could just be snake case instead.

Reply:
If that's a lot of work we can make a different story to do that